fixed 4 Test Classes

anuragSharma1112 opened the pull request:
# Description
Introduced an application-test.yml under src/test/resources to configure an in-memory SQLite database (jdbc:sqlite:memory:testdb?cache=shared) with Hibernate create-drop strategy and disabled Flyway migrations for test runs. This setup ensures isolated, lightweight database testing without external dependencies.
Also remove the earlier saperate Wiremock Service Impl and have created a test class for the WireMock Integration Test with Graph Email.

Fixes # (issue)
Test Classes Fixed

Please delete options that are not relevant.

## How Has This Been Tested?
- [X] Unit Tests
- [X] Integration Tests
- [ ] Manual Testing (please describe)

## Checklist:
- [ ] I have created ADR (Architecture Decision Record) for this change, in case of architecture or tech stack changes.
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes in other branches have been merged
- [X] I have updated the relevant application properties files (if needed)
- [X] I have updated any necessary database migrations (if needed)

## Screenshots (for UX changes):
NA

## Additional Notes:
All test classes are passed and build is green now, as i have created a application-test.yml and also handle the profiles and resolved the wiremock service as well and created a robust test class for wiremock Integration test with GraphEmail.

Commit:
fixed 4 Test Classes

anuragSharma1112 commented:
Hi @axymthr plz review

axymthr commented on web-backend/src/test/java/com/statusneo/vms/config/TestConfig.java:
```diff
@@ -0,0 +1,12 @@
+@TestConfiguration
+public class TestConfig {
+    @Bean
+    public RestTemplate restTemplate() {
```
Don't use RestTemplate

Commit:
add application-test.yml and fixed all test classes

anuragSharma1112 commented:
Hi @axymthr sir,
I corrected all failing test cases across the project. Since the build only passes when all related files are fixed together, pushing them in smaller batches (2–3 files at a time) would have caused the build to fail repeatedly.
To ensure a clean, successful CI/CD build, I fixed all affected files and committed them in one go.

axymthr committed:
Merge branch 'develop' into test-framework-fix

axymthr reviewed (approved):
PR looks good. The framework support is being used aptly. I think the Wiremock usage could be tweaked more so we don't have to do as much manual mocking, but let's do that in a separate PR.
